testing/wireguard-vanilla/APKBUILD

```diff
@@ -0,0 +1,67 @@
+# Contributor: Stuart Cardall <developer@it-offshore.co.uk>
+# Maintainer: Stuart Cardall <developer@it-offshore.co.uk>
+
+_flavor=${FLAVOR:-vanilla}
+_kpkg=linux-$_flavor
+_kver=4.4.47
+_kpkgrel=0
+
+# when changing _ver we *must* bump _mypkgrel
+_ver=0.0.20170223
+_mypkgrel=0
+_name=wireguard
+
+# verify the kernel version before entering chroot
+_kapkbuild=../../linux-${_flavor}/APKBUILD
+if [ -f $_kapkbuild ]; then
+	. $_kapkbuild
+	pkgname=$_name-$_flavor
+	[ "$_kver" != "$pkgver" ] && die "please update _kver to $pkgver"
+	[ "$_kpkgrel" != "$pkgrel" ] && die "please update _kpkgrel to $pkgrel"
+fi
+
+_kpkgver="$_kver-r$_kpkgrel"
+_abi_release=${_kver}
+
+pkgname=${_name}-${_flavor}
+pkgver=$_kver
+pkgrel=$(($_kpkgrel + $_mypkgrel))
+pkgdesc="Next generation secure network tunnel: kernel modules for $_flavor"
+arch='all'
+url='https://www.wireguard.io'
+license="GPLv2"
+makedepends="linux-vanilla-dev=$_kpkgver libmnl-dev sparse"
+source="https://git.zx2c4.com/WireGuard/snapshot/WireGuard-$_ver.tar.xz"
+builddir="$srcdir"/WireGuard-$_ver
+
+build() {
+	cd "$builddir"
+	# only building module: see wireguard-tools for userspace
+	make -C src/ \
+		KERNELDIR=/lib/modules/${_abi_release}/build \
+		module \
+		|| return 1
+}
+
+package() {
+	cd "$builddir/src"
+	install_if="linux-${_flavor}=${_kver} $pkgname"
+
+	local module=
+	for module in *.ko; do
+		install -v -D -m644 ${module} \
```

Comment:
module-install will respond to `INSTALL_MOD_PATH` being set, actually. So this is the right way to do it. In the next WireGuard release, I'll investigate wiring DESTDIR up to `INSTALL_MOD_PATH`, so that these can be unified. Thanks for pointing out the shortcomings to me.